content/en/blog/2022/k8s-otel-expose.md

```diff
@@ -0,0 +1,308 @@
+kind: OpenTelemetryCollector
+metadata:
+  name: otel-collector-app
+spec:
+  mode: deployment
+  image: otel/opentelemetry-collector-contrib:0.58.0
+  config: |
+    extensions:
+      basicauth/server:
+        htpasswd:
+          inline: |
+            <basic username>:<basic password>
+
+    receivers:
+      otlp/basicauth:
+        protocols:
+          grpc:
+            auth:
+              authenticator: basicauth/server
+
+    exporters:
+      otlp/jaeger:
+        endpoint: my-in-memory-collector:4317
+        tls:
+          insecure: true
+          insecure_skip_verify: true
+
+    service:
+      extensions: [basicauth/server]
+      pipelines:
+        traces:
+          receivers: [otlp/basicauth]
+          exporters: [otlp/jaeger]
+```
+
+After a successful installation, a pod for the Jaeger backend and the OpenTelemetry collector should be created in the selected namespace.
+
+```bash
+NAME                                            READY   STATUS    RESTARTS   AGE
+my-in-memory-6c5f5f87c5-rnp99                   1/1     Running   0          4m
+otel-collector-app-collector-55cccf4b7d-llczt   1/1     Running   0          3m
+```
+
+Also the following services should be available:
+```bash
+NAME                                      TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)                                                    AGE
+my-in-memory-agent                        ClusterIP   None            <none>        5775/UDP,5778/TCP,6831/UDP,6832/UDP                         7m
+my-in-memory-collector                    ClusterIP   10.245.43.185   <none>        9411/TCP,14250/TCP,14267/TCP,14268/TCP,4317/TCP,4318/TCP    7m
+my-in-memory-collector-headless           ClusterIP   None            <none>        9411/TCP,14250/TCP,14267/TCP,14268/TCP,4317/TCP,4318/TCP    7m
+my-in-memory-query                        ClusterIP   10.245.91.239   <none>        16686/TCP,16685/TCP                                         7m
+otel-collector-app-collector              ClusterIP   10.245.5.134    <none>        4317/TCP                                                    5m
+otel-collector-app-collector-headless     ClusterIP   None            <none>        4317/TCP                                                    5m
+otel-collector-app-collector-monitoring   ClusterIP   10.245.116.38   <none>        8888/TCP                                                    5m
+
+```
+
+Finally, cert-manager is configured to automatically request TLS certificates from [lets encrypt](https://letsencrypt.org/) and make it available to the Ingress TLS configuration. The following `ClusterIssuer` and `Ingress` entries expose the `otel-collector-app-collector` service. Note that `host` and `email` do only contain placeholder values.
+
+```yaml
+apiVersion: cert-manager.io/v1
+kind: ClusterIssuer
+metadata:
+  name: letsencrypt
+  namespace: cert-manager
+spec:
+  acme:
+    server: https://acme-v02.api.letsencrypt.org/directory
+    email: <your email>
```

Comment:
I don't know how this is usually done, but it would be helpful to point out all of the fields you need to replace, like "<your email" and "<your donain endpoint>".  In the latter case it might also be good to include an example, that way people would know if they need to specify the protocol, port, etc.

Reply:
tried to mention them a few lines above. What would help you to here? Something like `<replace someone@example.com>`, `<replace https://cluster-domain.com>`?

Reply:
Sorry, I missed that.

Reply:
is this fixed, if so please resolve